pandas/lib.pyx

```diff
@@ -1055,6 +1055,20 @@ def string_array_replace_from_nan_rep(ndarray[object, ndim=1] arr, object nan_re
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def object_array_decode_bytes(ndarray[object, ndim=1] arr, object encoding):
+    """Decode any instances of bytes to str in arr using the given encoding."""
```

Comment:
@TomAugspurger I do not have examples where I have combined bytes and str in a single column in the past, but it hardly seemed necessary to prevent it here and now.  `str.decode()` is not sufficient for another reason, anyway: a MultiIndex is treated as an array of tuples, not a 2D array, so there's no single Series to decode.